langchain/chains/conversation/memory.py

```diff
@@ -40,6 +46,88 @@ def _save_context(self, inputs: Dict[str, Any], outputs: Dict[str, str]) -> None
         self.buffer += "\n" + "\n".join([human, ai])
 
 
+class ConversationEntityMemory(Memory, BaseModel):
+    """Conversation entity search and store to memory."""
+
+    buffer: List[dict] = Field(default_factory=list)
+    docstore: InMemoryDocstore
+    llm: LLM
+    entity_extraction_prompt: BasePromptTemplate = ENTITY_EXTRACTION_PROMPT
+    entity_summarization_prompt: BasePromptTemplate = ENTITY_SUMMARIZATION_PROMPT
+    history_window_len: int = 5  #: :meta private:
+    dynamic_key_list: List[str] = ["history", "entities"]  #: :meta private:
+    entities_per_turn: List[str] = Field(default_factory=list)  #: :meta private:
+
+    @property
+    def dynamic_keys(self) -> List[str]:
+        """Will always return list of dynamic keys.
+
+        :meta private:
+        """
+        return self.dynamic_key_list
+
+    def _format_buffer_to_string(self) -> str:
+        section = self.buffer[-self.history_window_len :]
+        result = []
+        for line in section:
+            result.append("Human: {input}".format(input=line["input"]))
+            result.append("AI: {response}".format(response=line["response"]))
+        return "\n".join(result)
+
+    def _load_dynamic_keys(self, inputs: Dict[str, Any]) -> Dict[str, str]:
+        """Return history buffer with entity summaries."""
+        prompt_input_keys = list(set(inputs.keys()) - set(self.dynamic_keys))
+        assert len(prompt_input_keys) == 1, "One input key expected"
+        input = inputs[prompt_input_keys[0]]
+        full_conversation = self._format_buffer_to_string() + "\nHuman: " + input
```

Comment:
as a tech debt thing, everything where we do string manipulation here and in other memorty classes we should try to make a prompt template. cn you add a bit todo at top of file?